webrender/src/device/gfx/device.rs

```diff
@@ -3482,49 +3520,42 @@ impl<B: hal::Backend> Device<B> {
                     self.frame_fence[self.next_id].is_submitted = true;
 
                     // present frame
-                    swap_chain
+                    match swap_chain
                         .present(
                             &mut self.queue_group.queues[0],
                             self.current_frame_id as _,
                             Some(&self.render_finished_semaphore),
-                        )
-                        .is_err()
+                        ) {
+                            Ok(suboptimal) => {
+                                match suboptimal {
+                                    Some(_) => {
+                                        warn!("Suboptimal");
+                                        self.wait_for_resize = true;
+                                    },
+                                    None => {}
+                                }
+                            }
+                            Err(presenterr) => {
+                                match presenterr {
+                                    PresentError::OutOfDate => { warn!("PresentError : OutOfDate")},
+                                    PresentError::SurfaceLost(surf) => {warn!("PresentError : SurfaceLost => {:?}", surf)},
+                                    PresentError::DeviceLost(dev) => {warn!("PresentError : DeviceLost => {:?}", dev)},
+                                    PresentError::OutOfMemory(mem) => {warn!("PresentError : OutOfMemory => {:?}", mem)}
```

Comment:
Are these `{}`s necessary? If yes, add one space between them. Also add a `,` for the last line.